mobile/src/main/java/com/gelakinetic/mtgfam/fragments/dialogs/CardViewDialogFragment.java

```diff
@@ -97,7 +108,7 @@ public boolean onLongClick(View view) {
                 return dialog;
             }
             case GET_LEGALITY: {
-                if (getCardViewFragment().mFormats == null || getCardViewFragment().mLegalities == null) {
+                if (null == getCardViewFragment() || getCardViewFragment().mFormats == null || getCardViewFragment().mLegalities == null) {
```

Comment:
If we already checked if `getCardViewFragment()` was null up at line 78, do we need to check it for each of the cases also?

Reply:
If we already checked if `getCardViewFragment()` was null up at line 78, do we need to check it for each of the cases also?

Reply:
I didn't think so but Android Studio was still warning. Perhaps there's something multi-threaded happening where the return can change from the beginning of the function to here.

Reply:
This applies to most of your other comments as well.